nixos/modules/config/xdg/portals/wlr.nix

```diff
@@ -0,0 +1,69 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.xdg.portal.wlr;
+  settingsFormat = pkgs.formats.ini { };
+  configFile = settingsFormat.generate "xdg-desktop-portal-wlr.ini" cfg.settings;
+in
+{
+  meta = {
+    maintainers = with maintainers; [ minijackson ];
+  };
+
+  options.xdg.portal.wlr = {
+    enable = mkEnableOption ''
+      desktop portal for wlroots-based desktops
+
+      This will add the <package>xdg-desktop-portal-wlr</package> package into
+      the <option>xdg.portal.extraPortals</option> option, and provide the
+      configuration file
+    '';
+
+    package = mkOption {
+      description = "Package containing the wlr desktop portal";
+      type = types.package;
+      default = pkgs.xdg-desktop-portal-wlr;
+    };
+
+    settings = mkOption {
+      description = ''
+        Configuration for <package>xdg-desktop-portal-wlr</package>.
+
+        See <command>man 5 xdg-desktop-portal-wlr</command> for supported
```

Comment:
This likely will not be installed so I would just refer people to `xdg-desktop-portal-wlr(5)`.